src/ol/proj.js

```diff
@@ -625,6 +626,52 @@ export function fromUserExtent(extent, destProjection) {
   return transformExtent(extent, userProjection, destProjection);
 }
 
+/**
+ * Creates a safe coordinate transform function from a coordinate transform function.
+ * "Safe" means that it can handle wrapping of x-coordinates for global projections,
+ * and that coordinates exceeding the projection validity extent's range will be
+ * clamped to the validity range.
+ * @param {Projection} sourceProj Source projection.
+ * @param {Projection} destProj Destination projection.
+ * @param {function(import("./coordinate.js").Coordinate): import("./coordinate.js").Coordinate} forward Transform function (source to destiation).
+ * @param {function(import("./coordinate.js").Coordinate): import("./coordinate.js").Coordinate} inverse Transform function (destiation to source).
+ * @return {function(import("./coordinate.js").Coordinate): import("./coordinate.js").Coordinate} Safe transform function (source to destiation).
+ */
+export function createSafeCoordinateTransform(
+  sourceProj,
+  destProj,
+  forward,
+  inverse
+) {
+  return function (coord) {
+    let x, y, worldsAway;
+    if (sourceProj.canWrapX()) {
+      worldsAway = getWorldsAway(coord, sourceProj);
+      const sourceExtent = sourceProj.getExtent();
+      if (worldsAway && sourceExtent) {
+        x = coord[0] - worldsAway * getWidth(sourceExtent);
+      }
+    }
+    const destExtent = destProj.getExtent();
+    if (destExtent) {
+      const clampMin = inverse(destExtent.slice(0, 2));
+      const clampMax = inverse(destExtent.slice(2, 4));
```

Comment:
These may not actually represent the min/max y-values (right side could be lower than left, left higher than right, maybe bottom inverts to top).  Do we care?

Reply:
Right, that's a problem, e.g. with polar projections. I'm currently looking into how to solve this.

Reply:
Changed the code now so we only clamp source coordinates to the transformed destination extent when transforming the original coordinates fails.

Reply:
The new code also handles the case where bottom and top are inverted.